core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1692,64 +1712,66 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
     }
   }
 
-  private[zio] final case class BitChunk(bytes: Chunk[Byte], minBitIndex: Int, maxBitIndex: Int)
-      extends Chunk[Boolean] {
+  private[zio] sealed abstract class BitChunkBase[T](
+    chunk: Chunk[T],
+    val x: Int,
+    val x2: Int
+  ) extends Chunk[Boolean] {
     self =>
 
-    override val length: Int =
+    protected val minBitIndex: Int
+    protected val maxBitIndex: Int
+
+    val length: Int =
       maxBitIndex - minBitIndex
 
-    override def apply(n: Int): Boolean =
-      (bytes(n >> 3) & (1 << (7 - (n & 7)))) != 0
+    protected def elementAt(n: Int): T
+
+    protected def newBitChunk(chunk: Chunk[T], min: Int, max: Int): Chunk[Boolean]
 
-    override def drop(n: Int): BitChunk = {
+    override def drop(n: Int): Chunk[Boolean] = {
       val index  = (minBitIndex + n) min maxBitIndex
-      val toDrop = index >> 3
-      val min    = index & 7
+      val toDrop = index >> x
+      val min    = index & x2
       val max    = maxBitIndex - index + min
-      BitChunk(bytes.drop(toDrop), min, max)
+      newBitChunk(chunk.drop(toDrop), min, max)
+    }
+
+    override def take(n: Int): Chunk[Boolean] = {
+      val index  = (minBitIndex + n) min maxBitIndex
+      val toTake = (index + x2) >> x
+      newBitChunk(chunk.take(toTake), minBitIndex, index)
     }
 
     override def foreach[A](f: Boolean => A): Unit = {
-      val minByteIndex    = (minBitIndex + 7) >> 3
-      val maxByteIndex    = maxBitIndex >> 3
-      val minFullBitIndex = (minByteIndex << 3) min maxBitIndex
-      val maxFullBitIndex = (maxByteIndex << 3) max minFullBitIndex
+      if (this.isEmpty) return
```

Comment:
Would avoid using `return`.

Reply:
Actually, this was only for debugging and I forgot to remove it. So I'll do